iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -132,27 +132,34 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
-    TableIdentifier fromTableId = TableIdentifier.of(from.namespace().toString(), from.name());
-    Table fromTable = loadTable(fromTableId);
+    Table fromTable = loadTable(from);
     String tableClusterId = fromTable.properties().get(CatalogConstants.OPENHOUSE_CLUSTERID_KEY);
+
+    // Preserve existing case if databases are the same
+    String toDatabaseName =
+        from.namespace().toString().equalsIgnoreCase(from.namespace().toString())
```

Comment:
Looks like this is comparing same.
```suggestion
        fromTable.namespace().toString().equalsIgnoreCase(from.namespace().toString())
```

Reply:
Oops needed to compare from with to, thanks